File: AssetManager/UserInterface/Forms/AssetManager/MunisUserForm.vb
Public Class MunisUserForm

    Public ReadOnly Property EmployeeInfo As MunisEmployeeStruct
        Get
            Using Me
                If DialogResult = DialogResult.Yes Then
                    Return SelectedEmpInfo
                End If
                Return Nothing
            End Using
        End Get
    End Property

    Private SelectedEmpInfo As MunisEmployeeStruct
    Private Const intMaxResults As Integer = 50

    Sub New(parentForm As Form)
        InitializeComponent()
        Tag = parentForm
        Icon = parentForm.Icon
        ShowDialog(parentForm)
    End Sub

    Private Async Sub EmpNameSearch(Name As String)
        Try
            MunisResults.DataSource = Nothing
            Dim strColumns As String = "a_employee_number,a_name_last,a_name_first,a_org_primary,a_object_primary,a_location_primary,a_location_p_desc,a_location_p_short"
            Dim strQRY As String = "SELECT TOP " & intMaxResults & " " & strColumns & " FROM pr_employee_master WHERE a_name_last LIKE '%" & UCase(Name) & "%' OR a_name_first LIKE '" & UCase(Name) & "'"
            Dim MunisComms As New MunisComms
            SetWorking(True)
            Using results As DataTable = Await MunisComms.ReturnSqlTableAsync(strQRY)
                If results.Rows.Count < 1 Then Exit Sub
                MunisResults.DataSource = results
                MunisResults.ClearSelection()
            End Using
        Catch ex As Exception
            ErrHandle(ex, System.Reflection.MethodInfo.GetCurrentMethod())
        Finally
            SetWorking(False)
        End Try
    End Sub

    Private Sub SetWorking(Working As Boolean)
        pbWorking.Visible = Working
    End Sub

    Private Sub MunisUser_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        MunisResults.DefaultCellStyle = GridStyles
    End Sub

    Private Sub MunisResults_CellClick(sender As Object, e As DataGridViewCellEventArgs) Handles MunisResults.CellClick
        With SelectedEmpInfo
            .Name = MunisResults.Item(GetColIndex(MunisResults, "a_name_first"), MunisResults.CurrentRow.Index).Value.ToString & " " & MunisResults.Item(GetColIndex(MunisResults, "a_name_last"), MunisResults.CurrentRow.Index).Value.ToString.ToString
            .Number = MunisResults.Item(GetColIndex(MunisResults, "a_employee_number"), MunisResults.CurrentRow.Index).Value.ToString
        End With
        lblSelectedEmp.Text = "Selected Emp: " & SelectedEmpInfo.Name & " - " & SelectedEmpInfo.Number
    End Sub

    Private Sub cmdSearch_Click(sender As Object, e As EventArgs) Handles cmdSearch.Click
        Me.Cursor = Cursors.WaitCursor
        EmpNameSearch(Trim(txtSearchName.Text))
        Me.Cursor = Cursors.Default
    End Sub

    Private Sub SelectEmp()
        If SelectedEmpInfo.Name <> "" AndAlso SelectedEmpInfo.Number <> "" Then
            AssetFunc.AddNewEmp(SelectedEmpInfo)
            Me.DialogResult = DialogResult.Yes
            Me.Close()
        Else
            Me.DialogResult = DialogResult.Abort
            Me.Close()
        End If
    End Sub

    Private Sub cmdAccept_Click(sender As Object, e As EventArgs) Handles cmdAccept.Click
        SelectEmp()
    End Sub

    Private Sub txtSearchName_KeyDown(sender As Object, e As KeyEventArgs) Handles txtSearchName.KeyDown
        If e.KeyCode = Keys.Enter Then
            Me.Cursor = Cursors.WaitCursor
            EmpNameSearch(Trim(txtSearchName.Text))
            Me.Cursor = Cursors.Default
        End If
    End Sub

    Private Sub MunisResults_CellDoubleClick(sender As Object, e As DataGridViewCellEventArgs) Handles MunisResults.CellDoubleClick
        SelectEmp()
    End Sub

    Private Sub MunisUserForm_Shown(sender As Object, e As EventArgs) Handles Me.Shown
        txtSearchName.Focus()
    End Sub
End Class

File: AssetManager/UserInterface/Forms/AssetManagement/MunisUserForm.cs
using AssetManager.UserInterface.CustomControls;
using System;
using System.Data;
using System.Windows.Forms;

namespace AssetManager.UserInterface.Forms.AssetManagement
{
    public partial class MunisUserForm : ExtendedForm
    {
        public MunisEmployeeStruct EmployeeInfo
        {
            get
            {
                using (this)
                {
                    if (DialogResult == DialogResult.Yes)
                    {
                        return SelectedEmpInfo;
                    }
                    return new MunisEmployeeStruct();
                }
            }
        }

        private MunisEmployeeStruct SelectedEmpInfo = new MunisEmployeeStruct();

        private const int intMaxResults = 50;

        public MunisUserForm(ExtendedForm parentForm)
        {
            Shown += MunisUserForm_Shown;
            Load += MunisUser_Load;
            InitializeComponent();
            this.ParentForm = parentForm;
            ShowDialog(parentForm);
        }

        private async void EmpNameSearch(string Name)
        {
            try
            {
                MunisResults.DataSource = null;
                string strColumns = "a_employee_number,a_name_last,a_name_first,a_org_primary,a_object_primary,a_location_primary,a_location_p_desc,a_location_p_short";
                string strQRY = "SELECT TOP " + intMaxResults + " " + strColumns + " FROM pr_employee_master WHERE a_name_last LIKE '%" + Name.ToUpper() + "%' OR a_name_first LIKE '" + Name.ToUpper() + "'";
                MunisComms MunisComms = new MunisComms();
                SetWorking(true);
                using (DataTable results = await MunisComms.ReturnSqlTableAsync(strQRY))
                {
                    if (results.Rows.Count < 1)
                        return;
                    MunisResults.DataSource = results;
                    MunisResults.ClearSelection();
                }
            }
            catch (Exception ex)
            {
                ErrorHandling.ErrHandle(ex, System.Reflection.MethodInfo.GetCurrentMethod());
            }
            finally
            {
                SetWorking(false);
            }
        }

        private void SetWorking(bool Working)
        {
            pbWorking.Visible = Working;
        }

        private void MunisUser_Load(object sender, EventArgs e)
        {
            MunisResults.DefaultCellStyle = StyleFunctions.DefaultGridStyles;
        }

        private void MunisResults_CellClick(object sender, DataGridViewCellEventArgs e)
        {
            SelectedEmpInfo.Name = GridFunctions.GetCurrentCellValue(MunisResults, "a_name_first") + " " + GridFunctions.GetCurrentCellValue(MunisResults, "a_name_last");
            SelectedEmpInfo.Number = GridFunctions.GetCurrentCellValue(MunisResults, "a_employee_number");
            lblSelectedEmp.Text = "Selected Emp: " + SelectedEmpInfo.Name + " - " + SelectedEmpInfo.Number;
        }

        private void cmdSearch_Click(object sender, EventArgs e)
        {
            this.Cursor = Cursors.WaitCursor;
            EmpNameSearch(txtSearchName.Text.Trim());
            this.Cursor = Cursors.Default;
        }

        private void SelectEmp()
        {
            if (!string.IsNullOrEmpty(SelectedEmpInfo.Name) && !string.IsNullOrEmpty(SelectedEmpInfo.Number))
            {
                GlobalInstances.AssetFunc.AddNewEmp(SelectedEmpInfo);
                this.DialogResult = DialogResult.Yes;
                this.Close();
            }
            else
            {
                this.DialogResult = DialogResult.Abort;
                this.Close();
            }
        }

        private void cmdAccept_Click(object sender, EventArgs e)
        {
            SelectEmp();
        }

        private void txtSearchName_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                this.Cursor = Cursors.WaitCursor;
                EmpNameSearch(txtSearchName.Text.Trim());
                this.Cursor = Cursors.Default;
            }
        }

        private void MunisResults_CellDoubleClick(object sender, DataGridViewCellEventArgs e)
        {
            SelectEmp();
        }

        private void MunisUserForm_Shown(object sender, EventArgs e)
        {
            txtSearchName.Focus();
        }
    }
}